editoast/src/core/v2/conflict_detection.rs

```diff
@@ -0,0 +1,59 @@
+use chrono::DateTime;
+use chrono::Utc;
+use serde::Deserialize;
+use serde::Serialize;
+use std::collections::HashMap;
+use utoipa::ToSchema;
+
+use crate::core::{AsCoreRequest, Json};
+
+use super::simulation::RoutingRequirement;
+use super::simulation::SpacingRequirement;
+
+editoast_common::schemas! {
+    ConflictDetectionResponse,
+}
+
+#[derive(Debug, Serialize)]
+pub struct ConflictDetectionRequest {
+    /// List of requirements for each train
+    pub trains_requirements: HashMap<i64, TrainRequirements>,
+}
+
+#[derive(Debug, Serialize)]
+pub struct TrainRequirements {
+    pub start_time: DateTime<Utc>,
+    pub spacing_requirements: Vec<SpacingRequirement>,
+    pub routing_requirements: Vec<RoutingRequirement>,
+}
+
+#[derive(Debug, Deserialize, ToSchema)]
+pub struct ConflictDetectionResponse {
+    /// List of conflicts detected
+    #[schema(inline)]
+    pub conflicts: Vec<CoreConflict>,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize, ToSchema)]
+pub struct CoreConflict {
+    /// List of train ids involved in the conflict
+    pub train_ids: Vec<i64>,
+    /// Datetime of the start of the conflict
+    pub start_time: DateTime<Utc>,
+    /// Datetime of the end of the conflict
+    pub end_time: DateTime<Utc>,
+    /// Type of the conflict
+    #[schema(inline)]
+    pub conflict_type: ConflictType,
+}
+
+#[derive(Debug, Clone, Copy, Serialize, Deserialize, ToSchema)]
+pub enum ConflictType {
+    Spacing,
+    Routing,
```

Comment:
I'm wondering if it would be useful to actually document a bit more what 'Spacing' conflict or 'Routing' conflict means. Because 3 weeks ago, I would have found this code and would have had no idea what this means. Since this is part of the OpenAPI, documentation is probably a good idea?